package/src/components/Message/MessageSimple/MessagePinned.tsx

```diff
@@ -0,0 +1,149 @@
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends DefaultUserType = DefaultUserType,
+> = Pick<MessageContextValue<At, Ch, Co, Ev, Me, Re, Us>, 'alignment' | 'message'>;
+
+const MessagePinnedWithContext = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends DefaultUserType = DefaultUserType,
+>(
+  props: MessagePinnedPropsWithContext<At, Ch, Co, Ev, Me, Re, Us>,
+) => {
+  const { alignment, message } = props;
+  const {
+    theme: {
+      colors: { grey },
+    },
+  } = useTheme();
+  const { t } = useTranslationContext();
+  const { client } = useChatContext();
+  return (
+    <View
+      style={[
+        styles.view,
+        {
+          justifyContent: alignment === 'left' ? 'flex-start' : 'flex-end',
+        },
+      ]}
+      testID='message-pinned'
+    >
+      <Pin height={16} pathFill={grey} width={24} />
+      <Text style={[{ color: grey }, styles.text]}>
+        Pinned by{' '}
+        {message?.pinned_by?.id === client?.user?.id ? t('You') : message?.pinned_by?.name}
+      </Text>
+    </View>
+  );
+};
+
+const areEqual = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType,
+>(
+  prevProps: MessagePinnedPropsWithContext<At, Ch, Co, Ev, Me, Re, Us>,
+  nextProps: MessagePinnedPropsWithContext<At, Ch, Co, Ev, Me, Re, Us>,
+) => {
+  const { message: prevMessage } = prevProps;
+  const { message: nextMessage } = nextProps;
+  const messageEqual =
+    prevMessage.deleted_at === nextMessage.deleted_at &&
+    prevMessage.status === nextMessage.status &&
+    prevMessage.type === nextMessage.type &&
+    prevMessage.text === nextMessage.text &&
+    prevMessage.pinned === nextMessage.pinned;
+  if (!messageEqual) return false;
+  return true;
+};
+
+const MemoizedMessagePinned = React.memo(
+  MessagePinnedWithContext,
+  areEqual,
+) as typeof MessagePinnedWithContext;
+
+export type MessagePinnedProps<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends DefaultUserType = DefaultUserType,
+> = Partial<MessagePinnedPropsWithContext<At, Ch, Co, Ev, Me, Re, Us>>;
+
+export const MessagePinned = <
```

Comment:
I would suggest renaming this to more appropriate name. Maybe `MessagePinnedLabel` or `MessagePinnedHeader`. Because its not exactly the pinned message itself, its just a header. 

@madsroskar any better suggestions for name?

Reply:
@vishalnarkhede I'm kind of of the opinion that even just `Header` could work, since it'll be in the context of an import of `MessagePinned`. But if we want to keep MessagePinned as part of the name, I think I like `MessagePinnedHeader` more than label, as I'd think of label as connected to the pins themselves.

If there's naming crashes with `Header`, I'd expect the importer to give it an alias for their use case. But I'm not sure that's in line with code elsewhere in the SDK as of now, though